libs/cdk/src/lib/utils/tokens/viewport-size.observable.ts

```diff
@@ -0,0 +1,20 @@
+import { inject, InjectionToken } from '@angular/core';
+import { map, Observable } from 'rxjs';
+import { ViewportRuler } from '@angular/cdk/overlay';
+import { startWith } from 'rxjs/operators';
+
+/**
+ * Injection token for the viewport size observable.
+ * Emits the viewport size in pixels, which are outputted by the ViewportRuler
+ * from @angular/cdk/overlay.
+ */
+export const ViewportSizeObservable = new InjectionToken<Observable<number>>('ViewportSizeObservable', {
+    providedIn: 'root',
+    factory: () => {
+        const viewportRuler = inject(ViewportRuler);
+        return viewportRuler
+            .change(50)
+            .pipe(map((e) => (e.target as Window).innerWidth))
+            .pipe(startWith(window.innerWidth));
```

Comment:
In most cases I see this directive should be used, width of the parent container is much more preferable rather than window width

Reply:
Added that @N1XUS